File: Algorithms/search_sort/C++/insertionsort.cpp
#include <iostream>
#include <vector>

using namespace std;

void sort(vector<int>&);

int main(){

    return 0;
}

void sort(vector<int> &arr){

    int n = arr.size();
    for(int i = 1; i < n; ++i){
        int j = i - 1;
        int pivot = arr[i];
        while(j > 0 && arr[j] > pivot){
            arr[j + 1] = arr[j];
            --j;
        }
        arr[j + 1] = pivot;
    }
}

File: Algorithms/search_sort/Python/insertionsort.py
# I am not sure for this implementation
def sort_(array):
    n = len(array)
    if n <= 1:
        return array

    for i in range(1, n):
        pivot = array[i]
        j = i
        while j > 0 and pivot < array[j - 1]:
            array[j] = array[j - 1]
            j -= 1
        array[j] = pivot

    return array


# This one I am ok
def sort(arr):
    n = len(arr)
    for i in range(1, n):
        j = i - 1
        pivot = arr[i]
        while j >= 0 and arr[j] > pivot:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = pivot
